### ISSUE TITLE
an obscure constant that has explicit construction under binary expansion

### ISSUE THREAD

**Issue by farteryhr** [Mysticial/y-cruncher-Formulas#9]
(articles in Chinese)
- https://zhuanlan.zhihu.com/p/31547382
- https://zhuanlan.zhihu.com/p/31624394

a constant named by @MaigoAkisame after "wrongly implemented shuffle algorithm", but has explicit construction as binary expansion:

C=0.00_111_0000_11111_000000_11... (binary)

maybe, from ycruncher's perspective it's like Liouvielle's constant or whatever else, if just considered from the pattern. but as found by the author, it appears in various other researches, thus i think it's (a bit) more interesting.

at best it could be constructed in-memory even on-the-fly, then the main work is just the base conversion. so the question is, does the ycruncher custom formula support (fast) constructions like this?

**Comment by Mysticial** [comment]
y-cruncher can only do things that can be represented as a small and finite number of things from this list: http://www.numberworld.org/y-cruncher/guides/custom_formulas_list.html

**Comment by farteryhr** [comment]
> y-cruncher can only do things that can be represented as a small and finite number of things from this list: http://www.numberworld.org/y-cruncher/guides/custom_formulas_list.html

yes i'm aware of this list and i've used it when writing the plastic number user formula. just wondering if explicit direct digit construction will be supported in the future, since for some constant there might be no other method faster than this.

**Comment by Mysticial** [comment]
Unfortunately no, there are no plans right now to extend the functionality of the custom formulas. The list of more urgent things to do is already very long. So it's unlikely I'll be adding any significant new functionality here any time soon.

**Comment by Mysticial** [comment]
I recently looped back to the custom formulas because the "urgent" things are done. And I also needed a ton of refactoring to incorporate the new Log(2) formulas that Jorge Zuniga discovered.

Direct digit construction for binary digits seems like it doesn't need y-cruncher unless the purpose is to use y-cruncher for its binary -> base10 radix conversion.

For something like Liouvielle numbers, because the terms grow so quickly, you can probably get away with hard-coding a finite number of terms? y-cruncher does support powering so you can just evaluate them directly for a finite # of terms. Obviously very inefficient if you are constructing the number in the same base you want the digits in, but probably not that slow if you want it in a different base since most of the arithmetic would be needed anyway.

Beyond that, I'm not sure what else would be in scope. Since any new functionality would need a reasonable API to express it as.

**Comment by farteryhr** [comment]
the worst idea i could think of is another DSL for (binary) digit generation.. or some kind of for-loop, but regarding the immutability of nature of "variables" we currently have...
add a `"Mutation"` section into `Scope` that runs after initialization `"Locals"` and before the `"Formula"`, and ~the band begins to play 🎵~ `var if else for(each) getbit setbit` ?

---

btw another constant that might not worth a new issue but looks cool and like told by Ramanujan in a dream:
https://mathworld.wolfram.com/FreimansConstant.html
https://en.wikipedia.org/wiki/Markov_spectrum

**Comment by jzhansen** [comment]
Is there a series that represents Freiman's Constant?. If so, I can look
for an accelerated hypergeometric series for y-cruncher.

Jorge

**Comment by farteryhr** [comment]
> Is there a series that represents Freiman's Constant?. If so, I can look for an accelerated hypergeometric series for y-cruncher. Jorge El jue, 21 mar 2024 a las 6:42, 屁 ***@***.***>) escribió:

i'm afraid no guy has tried hypergeometric series on this and neither is my brain capable of doing this. i don't have a clue about how much faster it could be compared to the straightforward implementation (the only contribution of mine here was the `plastic number - native.cfg`), though i see there's `Sqrt(2) - Fast Series.cfg`.... haven't tried it yet, sorry.

> These square root constants are extremely fast to compute.
> -- formulas page on numberworld.org

waiting for mathematicians...

**Comment by Mysticial** [comment]
Yeah you're not beating Newton's Method with a hypergeometric series.

**Comment by jzhansen** [comment]
I see, Freiman's Constant is algebraic. Good point.

My Best

Jorge
